hack/jenkins/build.sh

```diff
@@ -62,7 +62,11 @@ git clean -fdx
 # docker images -q | xargs -r docker rmi
 
 # Build
-go run ./hack/e2e.go -v --build
+if [[ "${KUBE_FASTBUILD:-}" == "true" ]]; then
```

Comment:
It looks like things have been stirred up since I last looked over the build code and I'm not sure that `make release` will actually honor `KUBE_FASTBUILD`.

TBH I think we should get rid of `hack/jenkins/build.sh`. Whoever is calling it now should just call `make (quick-)release`. There's some release-infra logic in there right now, though.